### PR TITLE
Fix broken CI and unit tests

### PR DESCRIPTION
**Pull request by prettymuchbryce**
Fixes unit and end-to-end tests broken by our Tendermint/CometBFT fork.
* Fixes `TestEndBlockValidatorUpdatesResultingInEmptySet` to account for the new mempool calls added in `ApplyBlock`.
* Fixes nil pointer exceptions happening in cases where `cosmosTx.Body` was `nil`.
*  Including `cosmos-sdk` in this repository in [this](https://github.com/dydxprotocol/cometbft/commit/5582523887a416da2ba16e81634d441ae5bf46b6) commit modified a lot of dependencies due to the way that Go dependency resolution works (as Tendermint and CosmosSDK share a number of dependencies and Go uses [MVS](https://go.dev/ref/mod#minimal-version-selection:~:text=nothing%20requires%20them.-,Replacement,-The%20content%20of)). This was breaking some end-to-end tests due to a change in `tm-db` from from `v0.6.6` to `v0.6.7`. Specifically `tm-db` includes `rocksdb` as a dependency and `v0.6.7` seems to change `rocksdb` from `tecbot/gorocksdb` to `cosmos/gorocksdb` which seems to be a breaking change, incompatible with the way `rocksdb` is being installed in the end-to-end tests in this repository. As a solution here I added a replacement direction to use `tm-db` at `v0.6.6`, which is the version used upstream.
* Resets `KeepInvalidTxsInCache` to `false` in `DefaultMempoolConfig`, which was causing unit tests to fail. This value should was modified in [this](https://github.com/dydxprotocol/cometbft/commit/f6e3e9384a43bd59830bc3ff99e122bd1ff8bc81) commit, but instead should be modified in the configuration, so I did that [here](https://github.com/dydxprotocol/v4/pull/847) in `v4` instead.

### CODE REVIEW

File: mempool/dydx_helpers.go [hunk 1]
```diff
@@ -19,7 +19,8 @@ func IsClobOrderTransaction(
 		return false
 	}
 
-	if len(cosmosTx.Body.Messages) == 1 &&
+	if cosmosTx.Body != nil &&
```
Comment (prettymuchbryce) [inline mempool/dydx_helpers.go:22]: This was potentially a bug that could cause have caused panics.

**Review by BrendanChou — APPROVED** [review]

File: config/config.go [hunk 2]
```diff
@@ -759,14 +759,13 @@ func DefaultMempoolConfig() *MempoolConfig {
...
-		TTLNumBlocks: 0,
-		// Note this is true to avoid re-processing invalid order transactions more than once.
-		KeepInvalidTxsInCache: true,
+		Size:                  5000,
+		MaxTxsBytes:           1024 * 1024 * 1024, // 1GB
+		CacheSize:             10000,
+		MaxTxBytes:            1024 * 1024, // 1MB
+		TTLDuration:           0 * time.Second,
+		TTLNumBlocks:          0,
+		KeepInvalidTxsInCache: false,
```
Comment (ttl33) [inline config/config.go:768]: The PR description says "Resets KeepInvalidTxsInCache to true". I think the PR description was meant ot be "Reset ... to false". Is this the case?

File: mempool/v1/mempool.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/tendermint/tendermint/types"
 )
 
+// Test, remove this.
```
Comment (ttl33) [inline mempool/v1/mempool.go:22]: nit: did oyu mean to remove this line?

File: state/execution_test.go [hunk 4]
```diff
@@ -582,6 +584,11 @@ func TestEndBlockValidatorUpdatesResultingInEmptySet(t *testing.T) {
 		{PubKey: vp, Power: 0},
 	}
 
+	// dYdX fork: Apply block should lock/unlock the mempool.
```
Comment (ttl33) [inline state/execution_test.go:587]: for my own understanding, could you point me to where Lock/Unlock/FlushAppConn were added?

File: mempool/dydx_helpers.go (hunk 1 above)
Comment (ttl33) [reply mempool/dydx_helpers.go:22]: nice catch! in what case can the body be empty? is it possible to send an empty body tx?

File: go.mod [hunk 5]
```diff
@@ -285,4 +288,7 @@ require (
 
 retract [v0.35.0, v0.35.9] // See https://github.com/tendermint/tendermint/discussions/9155
 
-replace github.com/cosmos/cosmos-sdk v0.47.0-alpha2 => github.com/dydxprotocol/cosmos-sdk v0.47.0-alpha2-dydx-fork
+replace (
+	github.com/cosmos/cosmos-sdk v0.47.0-alpha2 => github.com/dydxprotocol/cosmos-sdk v0.47.0-alpha2-dydx-fork
+	github.com/tendermint/tm-db => github.com/tendermint/tm-db v0.6.6
```
Comment (ttl33) [inline go.mod:293]: for my own understanding, why do we need to pin this to a specific version?

File: mempool/dydx_helpers.go (hunk 1 above)
Comment (prettymuchbryce) [reply mempool/dydx_helpers.go:22]: I am not sure if it's possible in practice, but this was happening in unit tests.

File: config/config.go (hunk 2 above)
Comment (prettymuchbryce) [reply config/config.go:768]: Yes. Thanks, fixed.

File: state/execution_test.go (hunk 4 above)
Comment (prettymuchbryce) [reply state/execution_test.go:587]: Yes it's [this](https://github.com/dydxprotocol/cometbft/commit/4eea1903d5506da5eebf8f36045957c19da2d182) commit.

**Review by ttl33 — APPROVED** [review]

File: go.mod (hunk 5 above)
Comment (ttl33) [reply go.mod:293]: This question is answered in the PR description

**Pull request merged**